client/issue-lib.js

```diff
@@ -0,0 +1,146 @@
+(function() {
+'use strict';
+
+if (window.Issue) {
+  return;
+}
+
+// Colours are copied from:
+// https://www.google.com/design/spec/style/color.html#color-color-palette
+var _reviewLevelMetadata = {
+  'daily': {
+    query: {label: 'Update-Daily'},
+    outOfSLOColor: '#B71C1C',  // Red 900
+  },
+  'weekly': {
+    query: {label: 'Update-Weekly'},
+    outOfSLOColor: '#C62828',  // Red 800
+  },
+  'fortnightly': {
+    query: {label: 'Update-Fortnightly'},
+    outOfSLOColor: '#D32F2F',  // Red 700
+  },
+  'monthly': {
+    query: {label: 'Update-Monthly'},
+    outOfSLOColor: '#E53935',  // Red 600
+  },
+  'quarterly': {
+    query: {label: 'Update-Quarterly'},
+    outOfSLOColor: '#F44336',  // Red 500
+  },
+  'none (P0)': {
+    query: {label: 'Pri-0', '-has': 'update'},
+  },
+  'none (P1)': {
+    query: {label: 'Pri-1', '-has': 'update'},
+  },
+  'none (P2)': {
+    query: {label: 'Pri-2', '-has': 'update'},
+  },
+  'none (P3)': {
+    query: {label: 'Pri-3', '-has': 'update'},
+  },
+};
+var _defaultReviewLevel = 'none';
+var _inSLOColor = '#4CAF50';  // Green 500
+var _noSLOColor = '#757575';  // Grey 600
+
+// This class uses C style programming with JSON objects (representing issues)
+// and static methods.
+// This is to maintain the ability to clone the object in chromez-behaviours.js.
```

Comment:
True, this was a significant refactor to the Issue class that I should have included it in the design doc. I wouldn't have blocked coding it up on a proposal review though, a mechanical restructure of a single class takes <30 minutes while design review cycles are significantly longer than that.

I'll update design docs with refactor plans before starting them in future.